src/build/env.py

```diff
@@ -23,8 +23,13 @@
 else:
     from importlib import metadata
 
+
 try:
-    import virtualenv
+    virtualenv_version = packaging.version.Version(metadata.version('virtualenv'))
+    if virtualenv_version < packaging.version.Version('20.0.35'):
+        virtualenv = None
+    else:
+        import virtualenv
 except ImportError:  # pragma: no cover
```

Comment:
This will be a `importlib.metadata.PackageNotFoundError`, but that has `ImportError` in its mro, and if anything really bad happened and `import virtualenv` failed via `ImportError`, this would still catch it, I suppose? it could be narrowed down if preferred. Not sure why there are pragma no covers here, seems like this could have been testable.

Reply:
This is only the case in importlib_metadata 3 which dropped support Python 2.7 or 3.5, and which we continue to support.

```
In [1]: !python -m pip install -U 'importlib_metadata==2.*'
Collecting importlib_metadata==2.*
  Downloading importlib_metadata-2.1.1-py2.py3-none-any.whl (10 kB)
Collecting zipp>=0.5
  Downloading zipp-3.4.0-py3-none-any.whl (5.2 kB)
Installing collected packages: zipp, importlib-metadata
Successfully installed importlib-metadata-2.1.1 zipp-3.4.0

In [2]: import importlib_metadata

In [3]: try:
   ...:     importlib_metadata.version('foo')
   ...: except Exception as e:
   ...:     e_ = e
   ...:

In [4]: e_
Out[4]: importlib_metadata.PackageNotFoundError('foo')

In [5]: type(e_).__mro__
Out[5]:
(importlib_metadata.PackageNotFoundError,
 FileNotFoundError,
 OSError,
 Exception,
 BaseException,
 object)
```

I suggest restructuring it as:

```py
try:
    import virtualenv
except ImportError:
    virtualenv = None
else:
    virtualenv_version = packaging.version.Version(metadata.version('virtualenv'))
    if virtualenv_version < packaging.version.Version('20.0.35'):
        virtualenv = None
```

This check should not be performed at import time.

Reply:
I've cached it on the env builder. I think it should have been cached across calls to IsolatedEnvBuilder though, looking at the usage in `__main__`.

Reply:
Is d52edca9151da58da2ac89219638bd7355da949c more like what you were thinking?